prov/verbs/configure.m4

```diff
@@ -19,7 +19,28 @@ AC_DEFUN([FI_VERBS_CONFIGURE],[
 				[],
 				[$verbs_PREFIX],
 				[$verbs_LIBDIR],
-				[verbs_ibverbs_happy=1],
+				[
+				AC_MSG_CHECKING(if libibverbs is linkable by libtool)
+				file=conftemp.$$.c
+				rm -f $file conftemp
+				cat > $file <<EOF
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
```

Comment:
If you include `<infiniband/verbs.h>`, then you don't need the separate prototype.  The `AC_CHECK_LIB` test is intentionally very general -- you don't have to be so general here because the prior test proved the existence of `<infiniband/verbs.h>`.  So, you can either:

* Ditch the comment and leave the `char` declaration, or
* Ditch the comment and include `<infiniband/verbs.h>` (because you know it exists)

If you go with the latter, you will need to pass the correct arg types to `ibv_open_device()`.